### DISCUSSION TITLE
classes/class_dampedspringjoint2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#244] [User-contributed notes]
# classes/class_dampedspringjoint2d

Inherits: Joint2D< Node2D< CanvasItem< Node< Object A physics joint that connects two 2D physics bodies with a spring-like force. Description: A physics joint that connects two 2D physics bodies wi...

https://docs.godotengine.org/en/stable/classes/class_dampedspringjoint2d.html

**Comment by TheoMangos** [comment]
For anyone extremely confused check these two links:
https://www.reddit.com/r/godot/comments/afe8ix/dynamically_instancing_joints/
https://github.com/godotengine/godot/issues/74720

Basically "Length" is the initial length and not the max length
If rest length is set to 0 it instead equals length
